host/bloodview/src/data-cal.c

```diff
@@ -198,45 +253,55 @@ static uint32_t data_cal__data_channel_to_acq_channel(
 void data_cal_fini(void *pw)
 {
 	struct data_cal_ctx *ctx = pw;
-
-	/* TODO: This assumes channels == sources, which means
-	 *       calibration does not yet work with flash mode.
-	 *
-	 * To do calibration correctly, we should split it into digital
-	 * and analogue calibration separatly:
-	 *
-	 * 1. Analog calibration for opamp offset and amplification (where available).
-	 *    We disable software oversampling and sample 16-bits for this calibration.
-	 *
-	 * 2. Apply analog calibration values.
-	 *
-	 * 3. Digital calibration for software offset and shift.
-	 *    We apply the user specified software oversample and collect 32-bit
-	 *    samples to determine the optimal shift and offset values.
-	 *
-	 * 4. Apply digital calibration values.
+	struct {
+		uint32_t gain;
+		uint32_t offset;
+		bool     set;
+	} source_opamp[BL_ACQ_SOURCE_MAX] = {{0, 0, false}};
+
+	/* As there might be multiple channels sharing the
+	 * same source, only the first channel is going to
+	 * set the opamp gain/offset for the source
 	 */
 	for (unsigned i = 0; i < ctx->count; i++) {
 		struct channel_data *c = &ctx->channel[i];
+		enum bl_acq_source src;
 		uint32_t channel;
 		uint32_t channel_shift;
 		uint32_t channel_offset;
-		uint32_t source_opamp_gain;
-		uint32_t source_opamp_offset;
 
-		channel = data_cal__data_channel_to_acq_channel(ctx, i);
+		channel = data_cal__data_channel_to_acq_channel(i);
 		if (channel == UINT32_MAX) {
 			continue;
 		}
 
-		data_cal__calibrate_channel(
+		src = c->src;
+		if (src >= BL_ACQ_SOURCE_MAX) {
+			continue;
+		}
+
+		uint32_t hw_scale =	main_menu_config_get_source_hw_oversample(src) -
+							main_menu_config_get_source_hw_shift(src);
```

Comment:
Indentation looks weird.

```suggestion
		uint32_t hw_scale = main_menu_config_get_source_hw_oversample(src) -
				main_menu_config_get_source_hw_shift(src);
```

Reply:
my local file seems to be OK, not sure if it's caused by editor setting or something...
will fix anyway